### PR TITLE
Implement new permissions system for Gdrive

### PR DESCRIPTION
**Pull request by lasryaric**

### CODE REVIEW

File: connectors/src/connectors/google_drive/index.ts [hunk 1]
```diff
@@ -304,43 +308,221 @@ export async function retrieveGoogleDriveConnectorPermissions(
...
+              const d = await driveClient.drives.get({
+                driveId: f.folderId,
+              });
+              fd.name = d.data.name;
+            }
+            return {
+              provider: c.type,
+              internalId: f.folderId,
+              parentInternalId: null,
+              type: "folder" as ConnectorResourceType,
```
Comment (fontanierh) [inline connectors/src/connectors/google_drive/index.ts:342]: Why is the `as` needed here ?

File: connectors/src/connectors/google_drive/index.ts [hunk 2]
```diff
@@ -304,43 +308,221 @@ export async function retrieveGoogleDriveConnectorPermissions(
...
-    where: {
-      connectorId: connectorId,
-    },
-  });
+      const driveClient = await getDriveClient(authCredentials);
 
-  const driveClient = await getDriveClient(c.connectionId);
+      const resources: ConnectorResource[] = await Promise.all(
+        folders.map((f) => {
+          return (async () => {
```
Comment (fontanierh) [inline connectors/src/connectors/google_drive/index.ts:325]: why do you need to return a self-executing async arrow function here ? Can't you make the `.map()` callback an async fn directly ?

**Review by fontanierh — DISMISSED** [review]
LGTM 🔥

File: connectors/src/connectors/google_drive/index.ts (hunk 1 above)
Comment (spolu) [reply connectors/src/connectors/google_drive/index.ts:342]: You can use `as const` i believe

File: connectors/src/connectors/google_drive/index.ts [hunk 3]
```diff
@@ -304,43 +308,221 @@ export async function retrieveGoogleDriveConnectorPermissions(
...
+            const folder = await driveClient.files.get({
```
[the remark this answers is not in the document]
Comment (spolu) [reply connectors/src/connectors/google_drive/index.ts:326]: It seems that this is for the title? We likely want to work to have it locally no?

File: connectors/src/connectors/google_drive/index.ts [hunk 4]
```diff
@@ -304,43 +308,221 @@ export async function retrieveGoogleDriveConnectorPermissions(
...
+              sourceUrl: null,
+              expandable:
+                (await GoogleDriveFiles.count({
+                  where: {
+                    connectorId: connectorId,
+                    parentId: f.driveFileId,
+                    mimeType: "application/vnd.google-apps.folder",
+                  },
+                })) > 0,
+              permission: (await GoogleDriveFolders.findOne({
```
Comment (spolu) [inline connectors/src/connectors/google_drive/index.ts:389]: I'm not sure I understand this one? how could this be `none` ?

File: connectors/src/connectors/google_drive/index.ts [hunk 5]
```diff
@@ -304,43 +308,221 @@ export async function retrieveGoogleDriveConnectorPermissions(
...
+): Promise<Result<void, Error>> {
+  const connector = await Connector.findByPk(connectorId);
+  if (!connector) {
+    return new Err(new Error(`Connector not found with id ${connectorId}`));
+  }
+
+  let shouldFullSync = false;
+  for (const [id, permission] of Object.entries(permissions)) {
+    shouldFullSync = true;
+    if (permission === "write") {
```
Comment (spolu) [inline connectors/src/connectors/google_drive/index.ts:504]: We should not support `write` or `read_write` we should error if we get these. It should be `read` or `none`

File: front/components/GoogleDriveFoldersPickerModal.tsx [hunk 6]
```diff
@@ -1,321 +0,0 @@
-import { Button } from "@dust-tt/sparkle";
```
Comment (spolu) [inline front/components/GoogleDriveFoldersPickerModal.tsx:1]: Wonderful!

File: front/pages/w/[wId]/ds/index.tsx [hunk 7]
```diff
@@ -398,14 +393,6 @@ export default function DataSourcesView({
         />
 
         <div>
-          {googleDrive && googleDrive.connector && (
```
Comment (spolu) [inline front/pages/w/[wId]/ds/index.tsx:401]: Yay!

File: front/pages/w/[wId]/ds/index.tsx [hunk 8]
```diff
@@ -398,14 +393,6 @@ export default function DataSourcesView({
...
-            <GoogleDriveFoldersPickerModal
-              owner={owner}
-              connectorId={googleDrive.connector.id}
-              isOpen={googleDrivePickerOpen}
```
Comment (spolu) [inline front/pages/w/[wId]/ds/index.tsx:405]: What about these two?

**Review by spolu — CHANGES_REQUESTED** [review]
Left a few questions and a few requests for change. Looks great!

File: connectors/src/connectors/google_drive/index.ts [hunk 9]
```diff
@@ -304,43 +308,212 @@ export async function retrieveGoogleDriveConnectorPermissions(
...
-  const folders = await GoogleDriveFolders.findAll({
-    where: {
-      connectorId: connectorId,
-    },
-  });
+      const driveClient = await getDriveClient(authCredentials);
+
+      const resources: ConnectorResource[] = await Promise.all(
+        folders.map(async (f): Promise<ConnectorResource> => {
+          const folder = await driveClient.files.get({
```
Comment (spolu) [inline connectors/src/connectors/google_drive/index.ts:325]: As per IRL, let's not got to GDrive here and show what we do have so look locally

File: front/pages/w/[wId]/ds/[name]/index.tsx [hunk 10]
```diff
@@ -408,6 +401,16 @@ function ManagedDataSourceView({
     throw new Error("Connector provider is not defined");
   }
 
+  useEffect(() => {
+    if (
+      typeof router.query.edit_permissions === "string" &&
+      router.query.edit_permissions === "true"
+    ) {
+      void router.push(`/w/${owner.sId}/ds/${dataSource.name}`);
```
Comment (spolu) [inline front/pages/w/[wId]/ds/[name]/index.tsx:409]: Probably add a small comment and probably an await?

**Review by spolu — CHANGES_REQUESTED** [review]
Left a few comments discussed IRL

+ let's clean-up unused routes :+1:

File: connectors/src/connectors/google_drive/index.ts (hunk 9 above)
Comment (lasryaric) [reply connectors/src/connectors/google_drive/index.ts:325]: Revised IRL conversation: we stick to Gdrive for now.

**Review by spolu — DISMISSED** [review]
LGTM

:100: redlines!

**Pull request merged**